### PR TITLE
add testing for ruby 2.3 && 2.4; dropping support for 1.9

### PR DESCRIPTION
**Pull request by majormoses**
other misc changes. 

https://github.com/sensu-plugins/sensu-plugins-feature-requests/issues/27

Signed-off-by: Ben Abrams <me@benabrams.it>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,8 +5,16 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 
 ## [Unreleased][unreleased]
```
Comment (eheydrick) [inline CHANGELOG.md]: Get rid of the second `[unreleased]`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,8 +33,9 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 ### Added
 - initial release
 
-[unreleased]: https://github.com/sensu-plugins/sensu-plugins-apache/compare/0.0.5...HEAD
+[unreleased]: https://github.com/sensu-plugins/sensu-plugins-apache/compare/1.0.0...HEAD
```
Comment (eheydrick) [inline CHANGELOG.md]: Capitalize unreleased to be consistent w/ other plugins.

File: sensu-plugins-apache.gemspec [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
-  require_relative 'lib/sensu-plugins-apache'
-end
+require_relative 'lib/sensu-plugins-apache'
 
 # pvt_key = '~/.ssh/gem-private_key.pem'
 
-Gem::Specification.new do |s|
+Gem::Specification.new do |s| # rubocop:disable Metrics/BlockLength
   s.authors                = ['Sensu-Plugins and contributors']
   # s.cert_chain             = ['certs/sensu-plugins.pem']
```
Comment (eheydrick) [inline sensu-plugins-apache.gemspec]: get rid of commented out cert junk.

File: sensu-plugins-apache.gemspec [hunk 4]
```diff
@@ -31,7 +29,7 @@ Gem::Specification.new do |s|
   s.platform               = Gem::Platform::RUBY
   s.post_install_message   = 'You can use the embedded Ruby by setting EMBEDDED_RUBY=true in /etc/default/sensu'
   s.require_paths          = ['lib']
-  s.required_ruby_version  = '>= 1.9.3'
+  s.required_ruby_version  = '>= 2.0.0'
   # s.signing_key            = File.expand_path(pvt_key) if $PROGRAM_NAME =~ /gem\z/
```
Comment (eheydrick) [inline sensu-plugins-apache.gemspec]: ditto.

File: Rakefile [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (eheydrick) [inline Rakefile:1]: Is this from rubocop? Looks like a good thing, we should add it to the skel if we're going to do it to all repos.

**Review by eheydrick — CHANGES_REQUESTED** [review]

File: Rakefile (hunk 5 above)
Comment (majormoses) [reply Rakefile:1]: ya its updated in the skel I think, pretty sure I upgraded all the rubocop stuff recently.

**Comment by majormoses** [comment]
@eheydrick updated

File: Rakefile (hunk 5 above)
Comment (majormoses) [reply Rakefile:1]: ya I found it: https://github.com/sensu-plugins/sensu-plugins-skel/pull/2

**Review by eheydrick — APPROVED** [review]

**Comment by eheydrick** [comment]
👍

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-apache/versions/1.0.0

**Pull request merged**